modules/workload-identity/main.tf

```
@@ -25,9 +29,11 @@ locals {
 }
 
 resource "google_service_account" "cluster_service_account" {
+  for_each = var.use_existing_gcp_sa ? [] : toset([local.gcp_given_name])
```

Count is preferred for conditionals.

```suggestion
  count = var.use_existing_gcp_sa ? 0 : 1
```

---

OK. In our own code we've been moving the direction: collections keyed by a name rather than an index seem to be more resilient to changes.

---

Yes, collections keyed by name are preferred when it's actually a collection. This isn't a collection, just a binary (either 0 or 1 instances).